### DISCUSSION TITLE
`object-curly-newline` fine-grained control

### DISCUSSION THREAD

**Discussion by Bosseskompis** [eslint/eslint#16481] [Q&A] [answered]
Is there a way to accomplish the following?

```javascript
// allow
const { foo, bar, baz } = obj

// allow
const {
    foo,
    bar,
    baz,
} = obj

// disallow
const {
   foo, bar, baz,
} = obj

// disallow
const {
    foo, bar,
    baz,
} = obj
```

**Comment by mdjermanovic** [comment, marked as answer]
> ```js
> // disallow
> const {
>    foo, bar, baz,
> } = obj
> ```

You can set [`{ "multiline": true }`](https://eslint.org/docs/latest/rules/object-curly-newline#multiline) to disallow this.

> ```js
> // disallow
> const {
>     foo, bar,
>     baz,
> } = obj
> ```

There is no core rule that enforces newlines between properties in object patterns. It would be something like [object-property-newline](https://eslint.org/docs/latest/rules/object-property-newline) but for object patterns.
